addons/crm/models/crm_lead.py

```diff
@@ -0,0 +1,1117 @@
+    opt_out = fields.Boolean(string='Opt-Out', oldname='optout',
+            help="If opt-out is checked, this contact has refused to receive emails for mass mailing and marketing campaign. "
+                    "Filter 'Available for Mass Mailing' allows users to filter the leads when performing mass mailing.")
+    lead_type = fields.Selection(
+            [('lead', 'Lead'), ('opportunity', 'Opportunity')],
+            string='Lead Type', index=True, required=True, default=lambda self: 'lead' if self.env['res.users'].has_group('crm.group_use_lead') else 'opportunity',
+            help="Lead Type is used to separate Leads and Opportunities", oldname='type')
+    priority = fields.Selection(crm_stage.AVAILABLE_PRIORITIES, string='Rating', index=True, default=lambda *a: crm_stage.AVAILABLE_PRIORITIES[0][0])
+    date_closed = fields.Datetime(string='Closed', readonly=True, copy=False)
+    stage_id = fields.Many2one('crm.stage', string='Stage', track_visibility='onchange', index=True,
+                        domain="['|', ('team_id', '=', False), ('team_id', '=', team_id)]", default=lambda self: self._get_default_stage_id())
+    user_id = fields.Many2one('res.users', string='Salesperson', index=True, track_visibility='onchange', default=lambda self: self.env.user)
+    referred = fields.Char(string='Referred By')
+    date_open = fields.Datetime(string='Assigned', readonly=True)
+    day_open = fields.Float(compute='_compute_day', string='Days to Assign', store=True)
+    day_close = fields.Float(compute='_compute_day', string='Days to Close', store=True)
+    date_last_stage_update = fields.Datetime(string='Last Stage Update', index=True, default=fields.Datetime.now)
+    date_conversion = fields.Datetime(string='Conversion Date', readonly=True)
+
+    # Messaging and marketing
+    message_bounce = fields.Integer('Bounce', help="Counter of the number of bounced emails for this contact")
+    # Only used for type opportunity
+    probability = fields.Float(group_operator="avg", default=lambda self: self._get_default_probability())
+    planned_revenue = fields.Float(string='Expected Revenue', track_visibility='always')
+    date_deadline = fields.Date(string='Expected Closing', help="Estimate of the date on which the opportunity will be won.")
+    # CRM Actions
+    next_activity_id = fields.Many2one("crm.activity", string="Next Activity", index=True)
+    date_action = fields.Date(string='Next Activity Date', index=True)
+    title_action = fields.Char(string='Next Activity Summary')
+
+    color = fields.Integer(string='Color Index', default=0)
+    partner_address_name = fields.Char(related='partner_id.name', string='Partner Contact Name', readonly=True)
+    partner_address_email = fields.Char(related='partner_id.email', string='Partner Contact Email', readonly=True)
+    company_currency = fields.Many2one(related='company_id.currency_id', string='Currency', readonly=True, relation="res.currency")
```

Comment:
change name company_currency_id as it's many2one field and make separate commit.